base/ca/src/main/java/org/dogtagpki/server/ca/v2/AgentCertRequestServlet.java

```diff
@@ -0,0 +1,319 @@
+                String message = "CA disabled: " + e.getMessage();
+                logger.error(message, e);
+                throw new ConflictingOperationException(message, e);
+
+            } catch (CAMissingCertException | CAMissingKeyException e) {
+                throw new ServiceUnavailableException(e.toString(), e);
+
+            } catch (EPropertyException e) {
+                logger.error("AgentCertRequestServlet: Unable to change request state: " + e.getMessage(), e);
+                throw new PKIException("Unable to change request state: " + e.getMessage(), e);
+
+            } catch (EProfileException e) {
+                String message = CMS.getUserMessage(request.getLocale(), "CMS_INTERNAL_ERROR") + ": " + e.getMessage();
+                logger.error(message, e);
+                throw new PKIException(message, e);
+
+            } catch (EBaseException e) {
+                String message = "Unable to change request state: " + e.getMessage();
+                logger.error(message, e);
+                throw new PKIException(message, e);
+
+            } catch (RequestNotFoundException e) {
+                String message = "Unable to change request state: " + e.getMessage();
+                logger.error(message, e);
+                throw e;
+            }
+
+        } else {
+            response.sendError(HttpServletResponse.SC_NOT_FOUND, request.getRequestURI());
+        }
+    }
+
+    private CertReviewResponse getRequestData(HttpServletRequest servletRequest, RequestId id) throws EBaseException {
+        CertReviewResponse info = null;
+        CAEngine engine = getCAEngine();
+        RequestRepository requestRepository = engine.getRequestRepository();
+        ProfileSubsystem ps = engine.getProfileSubsystem();
+        SecureRandom random = null;
+        if (engine.getEnableNonces()) {
+            JssSubsystem jssSubsystem = engine.getJSSSubsystem();
+            random = jssSubsystem.getRandomNumberGenerator();
+        }
+
+
+        Request request = requestRepository.readRequest(id);
+
+        if (request == null) {
+            return null;
+        }
+
+        String profileId = request.getExtDataInString(Request.PROFILE_ID);
+
+        Profile profile = ps.getProfile(profileId);
+        info = CertReviewResponseFactory.create(request, profile, null, servletRequest.getLocale());
+
+        if (random != null) {
+            // generate nonce
+            long n = random.nextLong();
+            logger.info("AgentCertRequestServlet: Nonce: {}", n);
+
+            // store nonce in session
+            Map<Object, Long> nonces = engine.getNonces(servletRequest, "cert-request");
+            nonces.put(info.getRequestId().toBigInteger(), n);
+
+            // return nonce to client
+            info.setNonce(Long.toString(n));
+        }
+        if (info == null) {
+            // request does not exist
+            throw new RequestNotFoundException(id);
+        }
```

Comment:
I think `CertReviewResponseFactory.create()` cannot return a `null` value so this check is not needed.

This code probably should be moved here:
```
CertReviewResponse req = getRequestData(request, id);
if (req == null) {
    // request does not exist
    throw new RequestNotFoundException(id);
}
out.println(req.toJSON());
```